File: backend/Backend-API/Models/DbModels/Dog.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using static Backend_API.Models.Enums.Gender;
using static Backend_API.Models.Enums.Size;

namespace Backend_API.Models.DbModels
{
    public class Dog
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        [ForeignKey("Owner")]
        public string OwnerId { get; set; }
        public ApplicationUser Owner { get; set; }
        public double Age { get; set; }
        public string ImageURL { get; set; }
        [Required]
        public string Breed { get; set; }
        public string Color { get; set; }
        public eSize Size { get; set; }
        [Required]
        public eGender Gender { get; set; }
        public string Information { get; set; }
        public bool IsVaccinated { get; set; } //חיסון
        public bool IsNeutered { get; set; } //סירוס
    }
}

File: frontend/App/models/Dog.js
export default class Dog {
    constructor(ownerId ,name, age, race, color, gender, size, area, checkBoxItems, information, imageBase64)
    {
        this.ownerId = ownerId;
        this.name = name;
        this.age= age; 
        this.race = race; 
        this.color = color; 
        this.gender = gender;
        this.size = size; 
        this.area = area; 
        this.isVaccinated = false;
        this.isNeutered = false;
        this.information = information;
        this.imageBase64 = imageBase64;

        if(checkBoxItems.length === 2){
            this.isVaccinated = true;
            this.isNeutered = true;
        }
        else if(checkBoxItems.length === 1){
            this.isVaccinated = checkBoxItems[0].value === 'Vaccinated';
            this.isNeutered = checkBoxItems[0].value === 'Neutered';
        }
    }
}